tests/cpp/helpers.cu

```diff
@@ -40,4 +47,24 @@ std::shared_ptr<DMatrix> RandomDataGenerator::GenerateDeviceDMatrix(bool with_la
       0, bins_);
   return m;
 }
+
+#if defined(XGBOOST_USE_RMM) && XGBOOST_USE_RMM == 1
+using cuda_mr_t = rmm::mr::cuda_memory_resource;
+using pool_mr_t = rmm::mr::pool_memory_resource<cuda_mr_t>;
+class RMMAllocator {
+ public:
+  std::unique_ptr<pool_mr_t> handle;
+};
+
+void DeleteRMMResource(RMMAllocator* r) {
+  delete r;
+}
+
+RMMAllocatorPtr SetUpRMMResource() {
```

Comment:
You might consider providing parameters (like initial pool size, enable logging) in this API. However that could be added later if and as needed.

Reply:
The `SetupRMMResource()` method is internal to the C++ test suite, so users won't be directly calling them. I added it because the C++ test suite has to run standalone and so needs to set its own RMM allocator.

Reply:
I renamed the function as `SetupRMMResourceForCppTests()` to clarify what it's intended for.

Reply:
Got it. Sorry for the misunderstanding.